connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java

```diff
@@ -281,9 +281,18 @@ private void readToLogEnd() {
             Iterator<Map.Entry<TopicPartition, Long>> it = endOffsets.entrySet().iterator();
             while (it.hasNext()) {
                 Map.Entry<TopicPartition, Long> entry = it.next();
-                if (consumer.position(entry.getKey()) >= entry.getValue())
+                TopicPartition topicPartition = entry.getKey();
+                Long endOffset = entry.getValue();
+                long lastConsumedOffset = consumer.position(topicPartition);
+                if (lastConsumedOffset >= endOffset) {
+                    log.trace("Reached end offset {} for {}", endOffset, topicPartition);
```

Comment:
given that the previous messages say "Reading to ..." maybe it would make sense to say:
```suggestion
                    log.trace("Read to end offset {} for {}", endOffset, topicPartition);
```